app/views/devise/registrations/new.html.slim

```diff
@@ -18,7 +18,7 @@
       .form-group
         = f.email_field :email, autofocus: true, class: 'form-control', placeholder: 'Introduce your Email', required: true
       .form-group
-        = f.password_field :password, class: 'form-control', placeholder: 'Select your password (8 characters min.)', required: true
+        = f.password_field :password, class: 'form-control', pattern: "^[a-zA-Z0-9]{#{Devise.password_length.begin},#{Devise.password_length.end}}$", placeholder: "Select your password (#{Devise.password_length.begin} characters min.)", required: true
```

Comment:
I would not restrict password allowed symbols at all, I am all in for enforcing a minimum length, but characters should be any. Here is an example: http://html5pattern.com/Passwords (`(?=^.{8,}$)((?=.*\d)|(?=.*\W+))(?![.\n])(?=.*[A-Z])(?=.*[a-z]).*$`). Something similar to that?